### PR TITLE
Add a header to go.work

### PR DESCRIPTION
**Pull request by thockin**
Missed this before

/kind cleanup

```release-note
NONE
```

### CODE REVIEW

File: hack/update-go-workspace.sh [hunk 1]
```diff
@@ -37,6 +37,11 @@ rm -f go.work go.work.sum
 
 # Generate the workspace.
 go work init
+(
+  echo "// This is a generated file. Do not edit directly."
+  echo
+  cat go.work
+) | sponge go.work
```
Comment (liggitt) [inline hack/update-go-workspace.sh:44]: is sponge standard across Linux / Darwin?

Comment (thockin) [reply hack/update-go-workspace.sh:44]: ohh, good question.  I think it is across Linux.  Pesky Mac.

Comment (liggitt) [reply hack/update-go-workspace.sh]: Actually, failed the verify job as well with "command not found"

File: hack/update-go-workspace.sh [hunk 2]
```diff
@@ -37,6 +37,12 @@ rm -f go.work go.work.sum
 
 # Generate the workspace.
 go work init
+(
```
Comment (dims) [inline hack/update-go-workspace.sh:40]: `sed -i -e '1i\// This is a generated file. Do not edit directly.\n' go.work` may be simpler? but this is good as well!

**Pull request merged**